runtime/queries/metricsview_aggregation.go

```diff
@@ -305,6 +310,40 @@ func (q *MetricsViewAggregation) rewriteToMetricsViewQuery(export bool) (*metric
 		}
 	}
 
+	// If there is only one time dimension and null fill is enabled, we set the spine to the time range
+	if q.FillMissing && len(computedTimeDims) > 0 {
+		if qry.Spine != nil {
+			// should we silently ignore instead of error ?
+			return nil, fmt.Errorf("cannot have both where and time spine")
+		}
+		if len(computedTimeDims) > 1 {
+			return nil, fmt.Errorf("cannot have multiple time spines")
+		}
+		if q.TimeRange == nil || q.TimeRange.Start == nil || q.TimeRange.End == nil {
+			return nil, fmt.Errorf("time range is required for null fill")
+		}
```

Comment:
I think it might still work if there's a relative time range (like `q.TimeRange.IsoDuration`), and this might be needed e.g. for alerts/reports that use a time spine.

It would be resolved to a fixed time range before the query is turned into an AST. It happens here: https://github.com/rilldata/rill/blob/1d51e8ca44aa5a762cdf9bccffc841e858d514c4/runtime/metricsview/executor_rewrite_time.go#L15